x-pack/plugins/apm/common/connections.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { AgentName } from '../typings/es_schemas/ui/fields/agent';
+
+export enum NodeType {
+  service = 'service',
+  backend = 'backend',
+}
+
+interface NodeBase {
+  id: string;
+}
+
+export interface ServiceNode extends NodeBase {
+  type: NodeType.service;
+  serviceName: string;
+  agentName: AgentName;
+  environment: string;
+}
+
+export interface BackendNode extends NodeBase {
+  type: NodeType.backend;
+  backendName: string;
+  spanType: string;
+  spanSubtype: string;
+}
+
+export type Node = ServiceNode | BackendNode;
+
+export interface ConnectionMetricItem {
```

Comment:
Should we really call it `Metric`? I remember there were a lot of discussions when I implemented the comparison feature that "metric" is already over-utilized.

Reply:
agree, will change to "stats"